tests/unit/lms/resources/lti_launch_test.py

```diff
@@ -233,6 +233,25 @@ def test_it_returns_group_names_based_on_context_titles(
         assert LTILaunchResource(pyramid_request).h_group_name == expected_group_name
 
 
+class TestHSectionGroupName:
+    @pytest.mark.parametrize(
+        "section_name,group_name",
+        [
+            ("Test Section", "Test Section"),
+            (" Test Section ", "Test Section"),
+            ("Test   Section", "Test   Section"),
+            ("Object Oriented Polymorphism 101", "Object Oriented Polymorp…"),
```

Comment:
There's something to be said for:

`("x" * 100, "x" * 24 + "…")`

As it explains the length restriction in numbers rather than requiring you to count letters